cogito/protocol.go

```diff
@@ -79,18 +114,33 @@ func (src Source) String() string {
 	fmt.Fprintf(&bld, "gchat_webhook:         %s\n", redact(src.GChatWebHook))
 	fmt.Fprintf(&bld, "log_level:             %s\n", src.LogLevel)
 	fmt.Fprintf(&bld, "context_prefix:        %s\n", src.ContextPrefix)
+	fmt.Fprintf(&bld, "chat_append_summary:   %t\n", src.ChatAppendSummary)
 	// Last one: no newline.
 	fmt.Fprintf(&bld, "chat_notify_on_states: %s", src.ChatNotifyOnStates)
 
 	return bld.String()
 }
 
-// redact returns a redacted version of s. If s is empty, it returns the empty string.
-func redact(s string) string {
-	if s != "" {
-		s = "***REDACTED***"
+// UnmarshalJSON is used to set some default values of the struct.
+// See https://www.orsolabs.com/post/go-json-default-values/
+func (src *Source) UnmarshalJSON(data []byte) error {
+	type source Source // Alias to avoid infinite loop.
+
+	// Set the default value before JSON parsing.
+	aux := source{
+		ChatAppendSummary: true,
 	}
-	return s
+	// Since we also want to enforce the parser to fail if it encounters unknown fields,
+	// we cannot use the customary json.Unmarshal(data, &aux) but we have to go through
+	// a json decoder :-/
+	dec := json.NewDecoder(bytes.NewReader(data))
+	dec.DisallowUnknownFields()
+	if err := dec.Decode(&aux); err != nil {
+		return err
+	}
+	*src = Source(aux)
```

Comment:
@marco-m-pix4d I have a stupid question regarding this line. 

Source is `type Source struct` and aux is of the same type, so doing `Source(aux)` is some kind of type checking?
Is this similar to `int(6)` or `int(6.3)`?

Reply:
In Go it is called a "type conversion", see for a simple example https://go.dev/tour/basics/13.
This is different from "type assert" (https://go.dev/tour/methods/15).

Also @odormond explanation above should help.

With 
```go
*src = aux
```
we would get
```
cannot use aux (variable of type source) as type Source in assignment
```